apps/vaporgui/MainForm.cpp

```diff
@@ -428,6 +435,30 @@ MainForm::~MainForm()
     // no need to delete child widgets, Qt does it all for us?? (see closeEvent)
 }
 
+template<class T> bool MainForm::isDatasetValidFormat(const std::vector<std::string> &paths) const
+{
+    T dc;
+    bool errReportingEnabled = Wasp::MyBase::EnableErrMsg(false);
+    int ret = dc.Initialize(paths);
+    Wasp::MyBase::EnableErrMsg(errReportingEnabled);
+    return ret == 0;
+}
+
+int MainForm::determineDatasetFormat(const std::vector<std::string> &paths, std::string *fmt) const
+{
+    if (isDatasetValidFormat<VDCNetCDF>(paths))
+        *fmt = "vdc";
+    else if (isDatasetValidFormat<DCWRF>(paths))
+        *fmt = "wrf";
+    else if (isDatasetValidFormat<DCMPAS>(paths))
+        *fmt = "mpas";
+    else if (isDatasetValidFormat<DCCF>(paths))
+        *fmt = "cf";
+    else
+        return -1;
```

Comment:
The convention in VAPOR is to return a boolean if the function does not call MyBase::SetErrMsg on failure.

Reply:
Done but this is not mentioned in the official Vapor coding conventions.

Reply:
See https://github.com/NCAR/VAPOR/wiki/Proper-error-log-and-report-within-VAPOR

Reply:
I missed that, sorry.